### PR TITLE
feat: add search functionality to header

### PR DESCRIPTION
**Pull request by migueloller**
## What/Why?

Wires up quick search with Soul header section.

### General Use

https://github.com/user-attachments/assets/fc9b0798-9775-4309-b4e5-8c8d0d0b739a

### Submitting a query

https://github.com/user-attachments/assets/702fb725-357f-4f67-8651-6f6c1aaec474

### Internationalized empty state and avoiding showing empty or error states while searching

https://github.com/user-attachments/assets/07c47ea8-0e18-45df-ba6c-0af143a05679

### CODE REVIEW

**Review by apledger — APPROVED** [review]
LGTM!

**Review by apledger — APPROVED** [review]
LGTM!

File: core/vibes/soul/primitives/navigation/index.tsx [hunk 1]
```diff
@@ -26,6 +27,7 @@ import { Image } from '~/components/image';
 import { Link } from '~/components/link';
 import { usePathname } from '~/i18n/routing';
 
+import { FormStatus } from '../../form/form-status';
 import { Price } from '../price-label';
 import { ProductCard } from '../product-card';
```
Comment (jorgemoya) [inline core/vibes/soul/primitives/navigation/index.tsx:32]: All these should import from `@/vibes/soul/`

Comment (migueloller) [reply core/vibes/soul/primitives/navigation/index.tsx:32]: Thanks for the callout @jorgemoya! I'll open a PR that fixes this

Comment (migueloller) [reply core/vibes/soul/primitives/navigation/index.tsx:32]: VIBES: https://github.com/makeswift/vibes/pull/412
Catalyst: https://github.com/bigcommerce/catalyst/pull/1766

**Pull request merged**